### PR TITLE
feat(transactions): prefill create type from selected list tab

### PR DESCRIPTION
**Pull request by iguliaev**
## Why

When creating a transaction from the Transactions page, users usually expect the new transaction type to match the currently selected tab (Earn/Spend/Save). Requiring manual type re-selection adds friction and slows data entry.

## What Changed

- Files or areas updated:
  - `apps/web-next/src/pages/transactions/list.tsx`
  - `apps/web-next/src/pages/transactions/create.tsx`
  - `apps/web-next/e2e/tests/transactions.spec.ts`
  - `docs/superpowers/specs/2026-07-05-transaction-create-type-prefill-design.md`
- New functionality added:
  - Transactions list Create button now navigates with explicit context query params:
    - `source=transactions-list`
    - `type=<selected tab type>`
  - Transaction create form now preselects `Type` only when that context is present and valid.
- Existing behavior changed:
  - Direct `/transactions/create` (or invalid query params) keeps Type unselected and still requires explicit user choice.

## Key Decisions

- Decision:
  - Use URL query parameters with an explicit source marker instead of router state or implicit heuristics.
- Reasoning:
  - Query params are explicit, refresh-safe, and align with existing location-synced list behavior.
- Alternatives considered:
  - Router state (less durable across refresh/share)
  - Transient/global storage (more complexity and less transparency)

## How This Affects the System

- User-facing changes:
  - Smoother create flow from Transactions tabs with less repetitive input.
- API changes:
  - None.
- Performance implications:
  - Negligible; only lightweight query param parsing on create page.
- Database changes:
  - None.
- Breaking changes:
  - None.

## Testing

- New tests added:
  - `create page preselects type from transactions list tab context`
  - `direct create page keeps type unselected`
  - `invalid create query params do not preselect type`
- Existing tests status:
  - Targeted transaction E2E selectors pass for this behavior.
- Manual testing performed:
  - Validated navigation context and prefill behavior through the create flow.
- Edge cases tested:
  - Missing/invalid query context does not prefill Type.
- Test files modified or added:
  - `apps/web-next/e2e/tests/transactions.spec.ts`

### CODE REVIEW

File: apps/web-next/src/pages/transactions/list.tsx [hunk 1]
```diff
@@ -110,7 +120,20 @@ export const TransactionList = () => {
...
 
   return (
-    <List>
+    <List
+      headerButtons={() => (
+        <Button
+          type="primary"
+          onClick={() =>
+            navigate(
+              `/transactions/create?source=transactions-list&type=${transactionType}`
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/transactions/list.tsx]: fixed

File: apps/web-next/src/pages/transactions/create.tsx [hunk 2]
```diff
@@ -13,6 +19,18 @@ import {
...
+  const initialType = useMemo(() => {
+    const source = searchParams.get("source");
+    const type = searchParams.get("type");
+    const validTypes = new Set<TransactionType>(Object.values(TRANSACTION_TYPES));
+
+    if (source !== "transactions-list") return undefined;
+    if (!type || !validTypes.has(type as TransactionType)) return undefined;
+
+    return type;
+  }, [searchParams]);
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/transactions/create.tsx]: fixed

File: apps/web-next/src/pages/transactions/create.tsx [hunk 3]
```diff
@@ -54,7 +76,12 @@ export const TransactionCreate = () => {
 
   return (
     <Create saveButtonProps={{ ...saveButtonProps, loading: isLoading }}>
-      <Form {...formProps} layout="vertical" onFinish={handleFinish}>
+      <Form
+        {...formProps}
+        initialValues={initialType ? { type: initialType } : undefined}
+        layout="vertical"
+        onFinish={handleFinish}
+      >
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/transactions/create.tsx:98]: Fixed. `TransactionCreate` now merges `formProps.initialValues` with the validated type prefill instead of replacing initial values. This preserves existing/default Refine-provided defaults and only overrides `type` when context is valid.

[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/transactions/create.tsx:98]: fixed

**Comment by iguliaev** [comment]
@copilot run e2e tests
